spyder/widgets/editor.py

```diff
@@ -274,6 +274,70 @@ def breakpoints_changed(self):
             self.save_breakpoints.emit(self.filename, repr(breakpoints))
 
 
+class TabSwitcherWidget(QListWidget):
+    """Show tabs in mru order and change between them."""
+    def __init__(self, parent, stack_history, tabs):
+        QListWidget.__init__(self, parent)
+        self.setWindowFlags(Qt.SubWindow | Qt.FramelessWindowHint)
+
+        self.editor = parent
+        self.stack_history = stack_history
+        self.tabs = tabs
+
+        self.setSelectionMode(QListWidget.SingleSelection)
+        self.itemActivated.connect(self.item_selected)
+
+        self.id_list = []
+        self.load_data()
+        size = CONF.get('main', 'completion/size')

```

Comment:
I think It's better in the center of the editor (I take inspiration from Kate), at the center you will have to move less the eyes than if it's at the top.

Or maybe something like 1/4(editor height) from top?